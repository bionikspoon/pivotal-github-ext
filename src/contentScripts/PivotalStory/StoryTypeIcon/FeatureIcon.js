/* @flow */
import React from 'react'

export default function FeatureIcon() {
  return [
    <path key={0} d="M0 0h24v24H0z" fill="none" />,
    <path
      key={1}
      d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
    />,
    <path key={2} d="M0 0h24v24H0z" fill="none" />,
  ]
}
