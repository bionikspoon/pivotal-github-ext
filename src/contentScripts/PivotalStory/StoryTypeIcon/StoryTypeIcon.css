.StoryTypeIcon__container {
  display: none;
}
