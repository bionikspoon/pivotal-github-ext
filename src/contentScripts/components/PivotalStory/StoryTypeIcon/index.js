/* @flow */
export { default } from './StoryTypeIcon'
