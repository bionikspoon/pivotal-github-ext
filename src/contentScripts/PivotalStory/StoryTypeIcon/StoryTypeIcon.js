/* @flow */
import React from 'react'
import BugIcon from './BugIcon'
import FeatureIcon from './FeatureIcon'
import './StoryTypeIcon.css'

type Props = {
  storyType: string,
  size?: number,
  fill?: string,
}

export default function StoryTypeIcon(props: Props) {
  const Icon = props.storyType === 'bug' ? BugIcon : FeatureIcon
  return (
    <svg
      fill={props.fill}
      height={props.size}
      viewBox="0 0 24 24"
      width={props.size}
      xmlns="http://www.w3.org/2000/svg"
      className="StoryTypeIcon__container"
    >
      <Icon />
    </svg>
  )
}
StoryTypeIcon.defaultProps = {
  fill: '#5282b0',
  size: 24,
}
