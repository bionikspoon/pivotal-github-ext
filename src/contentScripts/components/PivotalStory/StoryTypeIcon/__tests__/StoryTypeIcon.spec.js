import React from 'react'
import { shallow } from 'enzyme'
import StoryTypeIcon from '../StoryTypeIcon'

describe('components/StoryTypeIcon', () => {
  test('it renders correctly for bugs', () => {
    const props = {
      storyType: 'bug',
      size: 200,
      fill: '#111111',
    }
    const wrapper = shallow(<StoryTypeIcon {...props} />)

    expect(wrapper).toMatchSnapshot()
  })

  test('it renders correctly for features', () => {
    const props = {
      storyType: 'feature',
      size: 200,
      fill: '#111111',
    }
    const wrapper = shallow(<StoryTypeIcon {...props} />)

    expect(wrapper).toMatchSnapshot()
  })
})
