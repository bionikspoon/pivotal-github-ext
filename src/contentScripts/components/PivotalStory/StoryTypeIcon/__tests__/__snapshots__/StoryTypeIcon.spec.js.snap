// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/StoryTypeIcon it renders correctly for bugs 1`] = `
<svg
  className="StoryTypeIcon__container"
  fill="#111111"
  height={200}
  viewBox="0 0 24 24"
  width={200}
  xmlns="http://www.w3.org/2000/svg"
>
  <BugIcon />
</svg>
`;

exports[`components/StoryTypeIcon it renders correctly for features 1`] = `
<svg
  className="StoryTypeIcon__container"
  fill="#111111"
  height={200}
  viewBox="0 0 24 24"
  width={200}
  xmlns="http://www.w3.org/2000/svg"
>
  <FeatureIcon />
</svg>
`;
